chimg/chroot.py

```diff
@@ -119,6 +119,27 @@ def _snaps_base_install(self, snap_infos: Dict[str, SnapInfo]) -> Dict[str, Snap
             snap_infos[core] = self._snap_install(name=core, channel="stable")
         return snap_infos
 
+    def _snap_delete(self, snap: SnapInfo):
+        """
+        Delete the given snap from the filesystem
+        """
+        logger.info(f"deleting snap {snap.name} from filesystem ...")
+        assertions_dir = f"{self._ctx.chroot_path}/{SNAPD_SEED_DIR}/assertions"
+        assertion_files = glob.glob(f"{assertions_dir}/{snap.name}_*.assert")
+        if len(assertion_files) != 1:
+            # TODO: use a chimg specific exception here
+            raise RuntimeError(f"Multiple .assert files available for snap {snap.name} in {assertions_dir}")
+        assertion_file = assertion_files[0]
+        run_command(["rm", "-f", assertion_file])
```

Comment:
Nit: This should be done after the second check has passed, i.e. as

```
run_command(["rm", "-f", assertion_file, snap_file])
```

My gut feeling tells me that if we can we shouldn't do any side effect until all checks have passed.

Reply:
fixed.